### PR TITLE
Image, input and responsive design

### PR DESCRIPTION
**Pull request by saurabhchirde**

### CODE REVIEW

File: components/alerts/alert.css [hunk 1]
```diff
@@ -12,8 +12,22 @@
...
+.alert-success-btn,
+.alert-info,
+.alert-info-btn,
+.alert-warning,
+.alert-warning-btn,
+.alert-error,
+.alert-error-btn{
+  min-width: 18rem;
+  height: auto;
 }
```
Comment (zoyatanweer) [inline components/alerts/alert.css:31]: Maybe create a class 'alert' and write all these properties into it.

File: components/buttons/button.css [hunk 2]
```diff
@@ -58,14 +59,14 @@
...
-.label-icon-outlined-btn-sm,
-.label-icon-outlined-btn-lg {
+.primary-outline-btn-md,
+.primary-outline-btn-xsm,
+.primary-outline-btn-sm,
+.primary-outline-btn-lg,
+.label-icon-outline-btn-md,
+.label-icon-outline-btn-xsm,
+.label-icon-outline-btn-sm,
+.label-icon-outline-btn-lg {
```
Comment (zoyatanweer) [inline components/buttons/button.css:69]: create a class outline and add into it.

File: components/alerts/alert.css (hunk 1 above)
Comment (saurabhchirde) [reply components/alerts/alert.css:31]: These properties are combined to reduce the classes.

File: components/buttons/button.css (hunk 2 above)
Comment (saurabhchirde) [reply components/buttons/button.css:69]: If I create a class with the name 'outline' then, the user would have to specify one more class to create that component.

**Review by m26unkwn — COMMENTED** [review]
Good to go Saurabh

File: index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// hamburger menu
+// document.querySelector(".hamburger").addEventListener("click", () => {
+//     document.querySelector("nav").classList.toggle("hide");
+// })
+
+// document.querySelector(".nav-close-btn").addEventListener("click", () => {
+//     document.querySelector("nav").classList.toggle("hide");
+// })
+
```
Comment (amarphule) [inline index.js:9]: remove unnecessary comments

File: index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+// const mediaQueryMobile = window.matchMedia('(max-width: 700px)');
+// const mediaQueryDesktop = window.matchMedia('(min-width: 700px)');
+
+// if (mediaQueryMobile.matches) {
+//     document.querySelector("nav").classList.toggle("hide");
+// }
+
+// if (mediaQueryDesktop.matches) {
+//     document.querySelector("nav").classList.remove("hide");
+// }
```
Comment (amarphule) [inline index.js:88]: same as above

**Review by amarphule — CHANGES_REQUESTED** [review]
Your code looks well structured and you followed proper naming conventions. Just one suggestion remove commented code. You have done great job.

File: components/alerts/alert.css [hunk 5]
```diff
@@ -12,8 +12,22 @@
 .alert-error-btn {
   max-width: 25rem;
   border-radius: 4px;
-  margin: 1rem;
+  margin: 0.6rem;
   padding: 0.6rem 0.6rem 0.6rem 0.8rem;
```
Comment (manojsarna) [inline components/alerts/alert.css:16]: use rems only if the pixel value is greater than 10 px.

File: components/alerts/alert.css [hunk 6]
```diff
@@ -50,16 +64,25 @@
...
+.alert-info-btn,
+.alert-warning-btn,
+.alert-error-btn {
   display: flex;
   align-content: center;
   justify-content: space-between;
+  height: 2.6rem;
 }
 
-.alert-text {
```
Comment (manojsarna) [inline components/alerts/alert.css:61]: great photos btw :)

File: components/common_css/flex.css [hunk 7]
```diff
@@ -2,7 +2,24 @@
...
     flex-direction: row; 
+    flex-wrap: wrap;
+}
+
+.display-flex-col-center{
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-direction: column; 
+    flex-wrap: wrap;
```
Comment (manojsarna) [inline components/common_css/flex.css:22]: improve class naming conventions

File: index.js (hunk 3 above)
Comment (manojsarna) [inline index.js:9]: still remove commented code!!

File: index.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    navBtn[0].classList.remove("nav-active");
+    navBtn[1].classList.remove("nav-active");
+    navBtn[2].classList.remove("nav-active");
+    navBtn[3].classList.remove("nav-active");
+    navBtn[4].classList.remove("nav-active");
+    navBtn[5].classList.remove("nav-active");
+    navBtn[6].classList.add("nav-active");
+})
+
+// media query
```
Comment (manojsarna) [inline index.js:77]: please improve your js code. ( hint: forEach !!!)

File: style.css [hunk 9]
```diff
@@ -48,14 +50,30 @@ body {
...
 
+.nav-header-mobile {
+    display: none;
+}
+
+.hamburger {
+    grid-area: ham;
+    position: sticky;
+    top: 0.7rem;
+    left: 0.5rem;
```
Comment (manojsarna) [inline style.css:61]: good naming conventions!!

**Review by manojsarna — APPROVED** [review]
overall Good Design and Code. !! LGTM!!

GTM!!

File: index.js (hunk 3 above)
Comment (saurabhchirde) [reply index.js:9]: working on that functionality, so commented.

File: index.js (hunk 4 above)
Comment (saurabhchirde) [reply index.js:88]: working on that functionality, so commented.

File: index.js (hunk 3 above)
Comment (saurabhchirde) [reply index.js:9]: working on that functionality, so commented.

File: components/alerts/alert.css (hunk 6 above)
Comment (saurabhchirde) [reply components/alerts/alert.css:61]: :)

File: components/common_css/flex.css (hunk 7 above)
Comment (saurabhchirde) [reply components/common_css/flex.css:22]: Done

File: style.css (hunk 9 above)
Comment (saurabhchirde) [reply style.css:61]: Thank you

**Review by PuneetP16 — COMMENTED** [review]
GTM, Good implementation of future visions.

File: components/common_css/flex.css (hunk 7 above)
Comment (gauravsinhaweb) [inline components/common_css/flex.css:22]: you can make a common CSS class instead of repeating properties.

Nice work done. Good to merge.

**Review by gauravsinhaweb — COMMENTED** [review]
you can make a common CSS class instead of repeating properties.

ref: common css/ flex CSS. line no. 5-25

Nice work done. Good to merge.

**Comment by ashishkatiyar2404** [comment]
its great saurabh your code is readable and you well follow the rule Don't repeat yourself. Nice😁

**Review by j24m — COMMENTED** [review]
Looks beautiful , code is so readable. Great work.💯

**Pull request merged**